### PR TITLE
Blocks and cards with no class substitution (via media query)

### PR DESCRIPTION
**Pull request by UladBohdan**
* Blocks and cards are to be auto-adjustable; size parameter is optional to be passed into the card; media query handles most of it.
* "Articles by Tag" `tag.js` page is rewritten using Blocks (same technique as with the main page).

Known issues:
* Some scss code is duplicated between files; has not yet figured out how can I avoid that.
* Tag page does not behave well on positioning cards (Block D - three in row) if not complete. Not critical. Will fix later.
Screen Shot 2019-08-15 at 5 42 26 PM

### CODE REVIEW

**Comment by UladBohdan** [comment]
Pushed to dev.wir.by

File: components/articles/blocks/TagPageBlockB.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import PropTypes from 'prop-types';
+
+import ArticleCard from 'components/articles/cards/ArticleCard';
+
+const TagPageBlockB = ({ articles, layout }) => (
+  <div
+    className={`block block__no-background tag-page-block-b ${
+      layout === 'large-left'
+        ? 'tag-page-block-b__style-large-left'
+        : 'tag-page-block-b__style-large-right'
```
Comment (Drapegnik) [inline components/articles/blocks/TagPageBlockB.js]: - `tag-page-block-b__style-${layout}`
- or `CLASS_BY_LAYOUT[layout]`

File: components/articles/blocks/TagPageBlockB.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }`}
+  >
+    <div className="large-card">
+      <ArticleCard {...articles[layout === 'large-left' ? 0 : 1]} />
+    </div>
+
+    <div className="small-card">
+      <ArticleCard {...articles[layout === 'large-left' ? 1 : 0]} />
```
Comment (Drapegnik) [inline components/articles/blocks/TagPageBlockB.js]:
```js
const [first, second] = layout === 'large-left' ? articles : articles.reverse()
```

```jsx
<ArticleCard {...first} />
```

File: components/articles/blocks/TagPageBlockCD.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ArticleCard from 'components/articles/cards/ArticleCard';
+
+const TagPageBlockCD = ({ articles, layout }) => (
+  <div
+    className={`block block__no-background tag-page-block-cd ${
+      layout === 'row-of-two'
+        ? 'tag-page-block-cd__style-row-of-two'
```
Comment (Drapegnik) [inline components/articles/blocks/TagPageBlockCD.js]: same as above

File: components/articles/blocks/TagPageBlockCD.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        : 'tag-page-block-cd__style-row-of-three'
+    }`}
+  >
+    {articles.slice(0, layout === 'row-of-two' ? 2 : 3).map(article => (
```
Comment (Drapegnik) [inline components/articles/blocks/TagPageBlockCD.js]:
```suggestion
    {articles.slice(0, SIZE_BY_LAYOUT[layout]).map(article => (
```

File: components/articles/cards/ArticleCard.js [hunk 5]
```diff
@@ -101,28 +83,25 @@ const ArticleCard = props => {
...
-                  {size === 'm' && article}
-                </span>
-              )}
-            />
-          </div>
-        )}
+        </div>
+        <div className="article__content-bottom">
+          <span className="article__description">{description}</span>
+          <span className={`article__label-read ${linkCn({ dark })}`}>
```
Comment (Drapegnik) [inline components/articles/cards/ArticleCard.js]: should work:
```suggestion
          <span className={linkCn({ className: 'article__label-read', dark })}>
```

File: components/articles/cards/CardWrapper.js [hunk 6]
```diff
@@ -6,9 +6,9 @@ import Link from 'components/common/Link';
 import BgContainer from 'components/common/ui/BgContainer';
 
 export const DEFAULT_SIZES = ['xxl', 'xl', 'l', 'm'];
-export const TAG_SIZES = ['s', 'xs'];
+export const TAG_SIZES = ['s', 'xs', 'ms', 's-wide', 'xs-wide'];
 export const SQUARE_SIZES = ['square-m', 'square-s'];
-export const SIZES = DEFAULT_SIZES.concat(SQUARE_SIZES, TAG_SIZES);
+export const SIZES = DEFAULT_SIZES.concat(SQUARE_SIZES, TAG_SIZES, 'auto');
```
Comment (Drapegnik) [inline components/articles/cards/CardWrapper.js:11]: 👍

File: constants/misc.js [hunk 7]
```diff
@@ -1,8 +1,8 @@
 import createConstants from 'lib/utils/createConstants';
 
-import TOPICS from 'constants';
+import { TOPICS } from 'constants';
 
-export const TOPIC = createConstants(TOPICS);
+export const TOPIC = createConstants(...TOPICS);
```
Comment (Drapegnik) [inline constants/misc.js:5]:
```suggestion
export const TOPIC = createConstants(TOPICS);
```

File: pages/markup/example2.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import 'components/articles/blocks/articlesByTag2.scss';
```
Comment (Drapegnik) [inline pages/markup/example2.js]: still need it?

File: pages/markup/mocks.js [hunk 9]
```diff
@@ -0,0 +1,3281 @@
+export default {
```
Comment (Drapegnik) [inline pages/markup/mocks.js]: don't merge mock file to base branch

File: pages/tag.js [hunk 10]
```diff
@@ -1,115 +1,57 @@
...
-      )}
-    </div>
-  );
-};
+const PAGE_LEVEL_ORDER = ['B1', 'C', 'D', 'C', 'B2', 'C', 'D', 'C'];
 
-const TagArticlesBlock = ({ block, invert, screen }) => {
-  const [levelB, levelC1, levelD, levelC2] = block;
+const TagArticles = ({ blocks }) => {
+  const data = blocks[0];
```
Comment (Drapegnik) [inline pages/tag.js]:
```suggestion
  const [data] = blocks;
```

File: pages/tag.js [hunk 11]
```diff
@@ -1,115 +1,57 @@
...
-      <TagLevel type="c" articles={levelC2} screen={screen} />
+      {data.map((_, index) => {
+        const levelName = PAGE_LEVEL_ORDER[index % PAGE_LEVEL_ORDER.length];
+        return {
+          B1: <TagPageBlockB articles={data[index]} layout="large-left" />,
+          B2: <TagPageBlockB articles={data[index]} layout="large-right" />,
+          C: <TagPageBlockCD articles={data[index]} layout="row-of-three" />,
+          D: <TagPageBlockCD articles={data[index]} layout="row-of-two" />,
+        }[levelName];
+      })}
```
Comment (Drapegnik) [inline pages/tag.js:60]:
```js
const getBlock = level => (...) ? TagPageBlockB : TagPageBlockCD;

const LAYOUT_BY_LEVEL = {
	B1: 'large-left',
	...
	D: 'row-of-two'
}
```

```jsx
return data.map((articles, index) => {
	const level = PAGE_LEVEL_ORDER[index % PAGE_LEVEL_ORDER.length];
	const Block = getBlock[level];
	return <Block key={index} layout={LAYOUT_BY_LEVEL[level]} />
});
```

**Review by Drapegnik — CHANGES_REQUESTED** [review]

**Comment by Drapegnik** [comment]
is `useComponentSize` unused now?

File: constants/misc.js (hunk 7 above)
Comment (UladBohdan) [reply constants/misc.js:5]: I remember fixing this, there was a bug causing TOPICS not to be exported at all. Why do you want to get rid of `...`?

File: pages/markup/example2.js (hunk 8 above)
Comment (UladBohdan) [reply pages/markup/example2.js]: Let's keep this as a sample.

File: pages/markup/mocks.js (hunk 9 above)
Comment (UladBohdan) [reply pages/markup/mocks.js]: Won't keeping it makes further development simpler?..

**Comment by UladBohdan** [comment]
[to rewrite Tag (Articles by Tag) page with Blocks; to introduce new blocks similar to those for the Main page](https://trello.com/c/MMrZDtGg/376-to-rewrite-tag-articles-by-tag-page-with-blocks-to-introduce-new-blocks-similar-to-those-for-the-main-page)

File: constants/misc.js (hunk 7 above)
Comment (Drapegnik) [reply constants/misc.js:5]: seems to be redundant

**Review by Drapegnik — APPROVED** [review]
huge work! 💪

**Pull request merged**